### ISSUE TITLE
Unable to use WSGI process with Raspbian

### ISSUE THREAD

**Issue by igorcmoura** [piku/piku#196]
Hi there!
I followed the [documentation on how to install piku for Raspbian](https://github.com/piku/piku/blob/master/docs/raspbian-9.4-stretch-10-buster.md) and it worked fine, until I tried to deploy an application with a WSGI process. It gives me an error saying `The -s/--socket option is missing and stdin is not a socket.`
Deploying the [sample application](https://github.com/piku/sample-python-app) gives me the same error. Here's the complete log:
```
wsgi.1 | *** Starting uWSGI 2.0.18-debian (32bit) on [Thu Jun 24 13:25:35 2021] ***
wsgi.1 | compiled with version: 8.2.0 on 10 February 2019 02:42:46
wsgi.1 | os: Linux-5.4.83-v7+ #1379 SMP Mon Dec 14 13:08:57 GMT 2020
wsgi.1 | nodename: raspberrypi
wsgi.1 | machine: armv7l
wsgi.1 | clock source: unix
wsgi.1 | pcre jit disabled
wsgi.1 | detected number of CPU cores: 4
wsgi.1 | current working directory: /home/piku/.piku/uwsgi-enabled
wsgi.1 | detected binary path: /usr/bin/uwsgi-core
wsgi.1 | setgid() to 33
wsgi.1 | setuid() to 1000
wsgi.1 | chdir() to /home/piku/.piku/apps/SampleWSGIApp
wsgi.1 | your processes number limit is 6878
wsgi.1 | your memory page size is 4096 bytes
wsgi.1 | detected max file descriptor number: 1024
wsgi.1 | lock engine: pthread robust mutexes
wsgi.1 | thunder lock: disabled (you can enable it with --thunder-lock)
wsgi.1 | Python version: 3.7.3 (default, Jan 22 2021, 20:04:44)  [GCC 8.3.0]
wsgi.1 | Set PythonHome to /home/piku/.piku/envs/SampleWSGIApp
wsgi.1 | Python main interpreter initialized at 0x20b1948
wsgi.1 | python threads support enabled
wsgi.1 | The -s/--socket option is missing and stdin is not a socket.
```
And here's the uWSGI configuration:
```
[uwsgi]
chdir = /home/piku/.piku/apps/SampleWSGIApp
uid = piku
gid = www-data
master = true
project = SampleWSGIApp
max-requests = 1024
listen = 16
processes = 1
procname-prefix = SampleWSGIApp:wsgi
enable-threads = true
log-x-forwarded-for = false
log-maxsize = 1048576
logfile-chown = piku:www-data
logfile-chmod = 640
logto2 = /home/piku/.piku/logs/SampleWSGIApp/wsgi.1.log
log-backupname = /home/piku/.piku/logs/SampleWSGIApp/wsgi.1.log.old
virtualenv = /home/piku/.piku/envs/SampleWSGIApp
module = main:app
threads = 4
plugin = python3
http = 0.0.0.0:9080
http-use-socket = 0.0.0.0:9080
log-format = %%(addr) - %%(user) [%%(ltime)] "%%(method) %%(uri) %%(proto)" %%(status) %%(size) "%%(referer)" "%%(uagent)" %%(msecs)ms
env = APP=SampleWSGIApp
env = LOG_ROOT=/home/piku/.piku/logs
env = HOME=/home/piku
env = USER=piku
env = PATH=/home/piku/.piku/envs/SampleWSGIApp/bin:/home/piku/.piku/envs/SampleWSGIApp/bin:/usr/lib/git-core:/usr/lib/git-core:/usr/lib/git-core:/home/piku/bin:/usr/local/sbin:/usr/sbin:/sbin:/usr/local/bin:/usr/bin:/bin:/usr/games
env = PWD=/home/piku/.piku/apps/SampleWSGIApp
env = VIRTUAL_ENV=/home/piku/.piku/envs/SampleWSGIApp
env = SETTING1=True
env = SETTING2=True/Maybe
env = PORT=9080
env = BIND_ADDRESS=0.0.0.0
env = RANGE=10
env = INTERVAL=1
env = NGINX_IPV4_ADDRESS=0.0.0.0
env = NGINX_IPV6_ADDRESS=[::]
env = PROC_TYPE=wsgi
```
I tried executing this configuration (just changing the paths) in my local Debian machine with both uWSGI from pip and from apt-get. It worked fine with pip but I got the same error with apt-get. It seems that the one from apt-get is unable to parse the `http` or `http-use-socket` configurations, and calling `uwsgi --help` also doesn't show these options.

So I guess the uWSGI from apt-get doesn't come with the http plugin (even though the [package description](https://packages.debian.org/stretch/uwsgi-core) says otherwise) but I couldn't find anyone with the same problem. Am I missing something?

**Comment by rcarmo** [comment]
Hi there!

That is odd, since Piku was developed on Raspbian and explicitly to work with the built-in packages (to minimize customisation). Can you be more specific about the version of Raspbian (`/etc/issue`) and share your `ENV` and `Procfile` (or sanitised equivalents)?

**Comment by chr15m** [comment]
@igorcmoura I think this was fixed with the PR to #204. If you can let us know if it works for you after an upgrade then we can close these two issues. Thanks!

**Comment by igorcmoura** [comment]
Yes! It fixed the problem!

Everything seems to be working fine now. Thanks! 🙂
